### DISCUSSION TITLE
Timer Events - Cron Expression

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [bpmnServer/bpmn-server#216] [Q&A]
Hi @ralphhanna,

Just wanted to clarify a quick thing about cron expression.

| Field| Required | Allowed values|Allowed special characters|
| :---------------- | :------: | ----: |----: |
|Seconds|  False| 0-59 | `*`  `,`  `-` |
| Minutes        |   True   | 0-59| `*` `,`  `-` |
| Hours           |   True   | 0-23 | `*` `,`  `-` |
| Day of month    |  True   | 1-31 | `*` `,`  `-` |
| Month |  True   | 1-12 | `*` `,`  `-` |
| Day of week|  True   | 0-6 | `*`  `,`  `-` |

I suppose the above would be the classic format supported in the Timer Definitions. Time is always in reference to UTC I suppose.
Correct me if I am wrong?

Does it also support the values `JAN–DEC` and `SUN-SAT` for Month & Day of Week? Also, are the special characters `L`, `W`, `?` & `#` supported?

**Comment by shashank-yelluri** [comment]
@MaheshkumarSundaram - if you've cracked the CRONs with this package, could you please share me the knowledge that you've obtained ? I've not explored much on the Timer Events. It'd be great if could share your knowledge.

Thank You

**Reply by MaheshkumarSundaram** [reply]
Sorry, I didn't look into this in detail as the existing support for Duration & Timer Cycles were sufficient for my usecases.

**Comment by ralphhanna** [comment]
Check out this help page
https://bpmnserver.github.io/docs/timers/

**Comment by ralphhanna** [comment]
Hi @MaheshkumarSundaram 

Sorry for taking so long to answer, I had to search it though.

The answer is yes, but I would recommend using the ISO 8601 format 'R3/PT25S' -- repeat 3 times 25 seconds delay
